include/libLiFFT/traits/IntegralType.hpp

```diff
@@ -28,6 +28,20 @@ namespace traits {
     template< typename T >
     struct IntegralType;
 
+
+
+  // SFINAE test if T has type member
+  template <typename T>
+  class has_type
```

Comment:
I have trouble understanding the usage of this. I assume this trait is supposed to return true iff a member `isComplex` exists, that is constructible from a int?
Then what is the reasoning in using it as used in `IntegralTypeImpl` below? If I read that correctly then `IntegralTypeImpl<Foo>` returns `Foo` for every type `Foo` that is either an int or float or simply does not have a `isComplex` member which is true for pretty much any class. Wouldn't that make it pointless?